src/www/ui/api/Controllers/JobController.php

```diff
@@ -148,16 +159,25 @@ public function getJobs($request, $response, $args)
    */
   public function createJob($request, $response, $args)
   {
-    $folder = $request->getHeaderLine("folderId");
-    $upload = $request->getHeaderLine("uploadId");
+    $apiVersion = ApiVersion::getVersion($request);
+    $folder = null;
+    $upload = null;
+    if ($apiVersion == ApiVersion::V2) {
+      $query = $request->getQueryParams();
+      $folder = $query["folderId"] ?? null;
+      $upload = $query["uploadId"] ?? null;
+    } else {
+      $folder = $request->hasHeader('folderId') ? $request->getHeaderLine('folderId') : null;
+      $upload = $request->hasHeader('uploadId') ? $request->getHeaderLine('uploadId') : null;
+    }
     if (is_numeric($folder) && is_numeric($upload) && $folder > 0 && $upload > 0) {
       $scanOptionsJSON = $this->getParsedBody($request);
       if (empty($scanOptionsJSON)) {
         throw new HttpBadRequestException("No agents selected!");
       }
       $uploadHelper = new UploadHelper();
       $info = $uploadHelper->handleScheduleAnalysis($upload, $folder,
-        $scanOptionsJSON);
+        $scanOptionsJSON, false, $apiVersion);
```

Comment:
Any reason for changing this?

Reply:
PTAL at https://github.com/fossology/fossology/pull/2736/files#diff-8a7e9933b8879b6ff5e18b4082859fccababcfdde56142701d2e83b298aa4a5cR115
If I don't pass `false`, `$newUpload` will take up the `$apiVersion`.